mlflow/projects/__init__.py

```diff
@@ -169,14 +169,20 @@ def _run(
             build_image=build_image,
             docker_auth=docker_auth,
         )
-        image_digest = kb.push_image_to_registry(image.tags[0])
+        if build_image:
+            kb.push_image_to_registry(image.tags[0], docker_auth)
+            image_tag = image.tags[0]
+        else:
+            image_tag = next(filter(lambda t: project.docker_env.get("image") in t, image.tags))
```

Comment:
Thanks for the PR! Is there any case where `project.docker_env.get("image")` doesn't match anything and `next` throws?

Reply:
```suggestion
            image_tag = next(filter(lambda t: project.docker_env.get("image") in t, image.tags), None)
            if image_tag is None:
                raise ...
```

Can we raise an exception if the specified image matches nothing?

Reply:
I don't think that line will fail because `build_docker_image` pulls `project.docker_env.get("image")`.

For example, if `project.docker_env.get("image")` is `ubuntu`, `image.tags` will be `[ubuntu:latest, ...]`
* https://github.com/mlflow/mlflow/blob/6fa2f725a5e6f8a5afab547a8f59e27b15791868/mlflow/projects/__init__.py#L164-L171
* https://github.com/mlflow/mlflow/blob/6fa2f725a5e6f8a5afab547a8f59e27b15791868/mlflow/projects/docker.py#L76-L83

Reply:
Ah that makes sense, thanks for the explanation!